### PR TITLE
refactor: capture network error + response time data

### PR DESCRIPTION
**Pull request by audrey-kho**
## Description

For each generate request, we will capture EIM data for:
- network errors
- response time

## Related Issue

 
https://jira.corp.adobe.com/browse/SITES-21134

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by askayastha22** [comment]
@audrey-kho Can you please finalize this PR?

**Comment by audrey-kho** [comment]
> @audrey-kho Can you please finalize this PR?

This was dependent on the other PR that contains the upgrade to `gpt-4-turbo` for PTU testing. I think it was decided in the most recent discussion that we won't be going with the upgraded version so does that mean we should just close this PR without merging?

File: web-src/src/components/GenerateButton.js [hunk 1]
```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
     sampleRUM('genai:prompt:generate', { source: 'GenerateButton#handleGenerate' });
     setGenerationInProgress(true);
     setIsOpenPromptEditor(false);
+    const startTime = Date.now();
     generateResults()
       .catch((error) => {
         ToastQueue.negative(error.message, { timeout: 2000 });
+
+        // capture network errors
+        if (error.status > 300) {
```
Comment (lenaorobei) [inline web-src/src/components/GenerateButton.js:83]: I'd separate [Client error responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) (400 – 499) and [Server error responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses) (500 – 599).

Comment (irenelagno) [reply web-src/src/components/GenerateButton.js:83]: +1

File: web-src/src/components/GenerateButton.js [hunk 2]
```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
...
+          log('prompt:generate:networkerror', { status: error.status, message: error.message });
+        }
       })
       .finally(() => {
         setGenerationInProgress(false);
+
+        const endTime = Date.now();
+        const responseTime = ((endTime - startTime) / 1000).toFixed(2);
```
Comment (irenelagno) [inline web-src/src/components/GenerateButton.js:91]: Please add comments like 
 const responseTime = ((endTime - startTime) / 1000).toFixed(2);// response time in seconds or use it in variable name like responseTimeSeconds

File: web-src/src/components/GenerateButton.js [hunk 3]
```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
...
+        log('prompt:generate:responsetime', { responseTime });
```
Comment (irenelagno) [inline web-src/src/components/GenerateButton.js:92]: make the message more specific. Add measurement units

**Review by irenelagno — CHANGES_REQUESTED** [review]

**Pull request closed without merging**